the semantics of slice patterns changed

https://github.com/rust-lang/rust/issues/23121

```
error[E0529]: expected an array or slice, found `&[&str]`
   --> src/lib.rs:380:9
    |
380 |         ["open", id] => {
    |         ^^^^^^^^^^^^
    |
    = help: the semantics of slice patterns changed recently; see issue #23121

error[E0529]: expected an array or slice, found `&[&str]`
   --> src/lib.rs:385:9
    |
385 |         ["closed", start] => {
    |         ^^^^^^^^^^^^^^^^^
    |
    = help: the semantics of slice patterns changed recently; see issue #23121

error: aborting due to 2 previous errors

error: Could not compile `wal`.
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -377,12 +377,12 @@ fn open_dir_entry(entry: fs::DirEntry) -> Result<WalSegment> {
 
     let filename = try!(entry.file_name().into_string().map_err(|_| error()));
     match &*filename.split('-').collect::<Vec<&str>>() {
-        ["open", id] => {
+        &["open", ref id] => {
             let id = try!(u64::from_str(id).map_err(|_| error()));
             let segment = try!(Segment::open(entry.path()));
             Ok(WalSegment::Open(OpenSegment { segment: segment, id: id }))
         },
-        ["closed", start] => {
+        &["closed", ref start] => {
             let start = try!(u64::from_str(start).map_err(|_| error()));
             let segment = try!(Segment::open(entry.path()));
             Ok(WalSegment::Closed(ClosedSegment { start_index: start, segment: segment }))

